spec/index.bs

```diff
@@ -5131,70 +5144,114 @@ enum GPUStoreOp {
 <dl dfn-type=method dfn-for=GPURenderEncoderBase>
     : <dfn>setPipeline(pipeline)</dfn>
     ::
+        Sets the current {{GPURenderPipeline}}.
 
         <div algorithm="GPURenderEncoderBase.setPipeline">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setPipeline(pipeline)">
-                pipeline:
+                |pipeline|: The render pipeline to use for subsequent drawing commands.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setPipeline()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |pipeline| is [=valid=].
+                        - |pipeline|.{{GPUObjectBase/[[device]]}} is |this|.
```

Comment:
future comment: These are two rules that we're going to use in a lot of places; eventually we should probably make a little helper algorithm for them e.g. "objectValidForDevice(object, device)"

Reply:
I thought the same thing, but wanted to keep any changes like that separate so it was easier to discuss how best to format them.